pkg/apis/migration/migration.go

```diff
@@ -407,3 +417,61 @@ func (volumeMigration *volumeMigration) registerVolume(ctx context.Context, volu
 	}
 	return volumeID.Id, nil
 }
+
+// cleanupStaleCRDInstances helps in cleaning up stale volume migration CRD instances
+func (volumeMigration *volumeMigration) cleanupStaleCRDInstances() {
+	ticker := time.NewTicker(time.Duration(volumeMigrationInstance.cnsConfig.Global.VolumeMigrationCRCleanupIntervalInMin) * time.Minute)
+	for range ticker.C {
+		ctx, log := logger.GetNewContextWithLogger()
+		config, err := k8s.GetKubeConfig(ctx)
+		if err != nil {
+			log.Warnf("failed to get kubeconfig. err: %v", err)
+			continue
+		}
+		crdGroupClient, err := k8s.NewClientForGroup(ctx, config, CRDGroupName)
+		if err != nil {
+			log.Warnf("failed to create new client for group %q . Err: %+v", CRDGroupName, err)
+			continue
+		}
+
+		log.Infof("Triggering CnsVSphereVolumeMigration cleanup routine")
+		volumeMigrationResourceList := &migrationv1alpha1.CnsVSphereVolumeMigrationList{}
+		err = crdGroupClient.List(ctx, volumeMigrationResourceList)
+		if err != nil {
+			log.Warnf("failed to get CnsVSphereVolumeMigration list from the cluster. Err: %+v", err)
+			continue
+		}
+		log.Debugf("CnsVSphereVolumeMigrationList: %+v", volumeMigrationResourceList)
+		queryFilter := cnstypes.CnsQueryFilter{
+			ContainerClusterIds: []string{
+				volumeMigrationInstance.cnsConfig.Global.ClusterID,
+			},
+		}
+		queryAllResult, err := (*volumeMigrationInstance.volumeManager).QueryAllVolume(ctx, queryFilter, cnstypes.CnsQuerySelection{})
+		if err != nil {
+			log.Warnf("failed to queryAllVolume with err %+v", err)
+			continue
+		}
+		log.Debugf("QueryVolumeInfo successfully returned with result:  %v:", spew.Sdump(queryAllResult))
+		cnsVolumesMap := make(map[string]bool)
+		for _, vol := range queryAllResult.Volumes {
+			cnsVolumesMap[vol.VolumeId.Id] = true
+		}
+		log.Debugf("cnsVolumesMap:  %v:", cnsVolumesMap)
+		var resourcesToBeDeleted []string
+		for _, volumeMigrationResource := range volumeMigrationResourceList.Items {
+			if _, existsInCNSVolumesMap := cnsVolumesMap[volumeMigrationResource.Name]; !existsInCNSVolumesMap {
+				log.Debugf("Volume with id %s is not found in CNS", volumeMigrationResource.Name)
+				resourcesToBeDeleted = append(resourcesToBeDeleted, volumeMigrationResource.Name)
```

Comment:
This is probably a nit but why not just delete the instance here?

Reply:
Actually it makes sense to delete right there. Thanks